### PR TITLE
Hardware accelerated transforms

### PR DESCRIPTION
**Pull request by MattiaVerticchio**
Hi, first of all, thanks for developing and open sourcing this package! 😁 

I made some changes on the generated CSS that allow using the hardware acceleration of the GPU for `translate`, `scale` and `rotate`. Also, I ran some `elm-review` rules.

### CODE REVIEW

File: src/Internal/Animation.elm [hunk 1]
```diff
@@ -20,8 +19,11 @@ import Internal.Unit as Unit
 -- Animation
 
 
-type Animation
-    = Animation Millis (List Option) (List Frame)
+type alias Animation =
```
Comment (andrewMacmurray) [inline src/Internal/Animation.elm:22]: Happy for the parameters to be simplified to a record 👍 , but this change means the internal record is exposed to users of the package. Can you make sure the internals aren't exposed? e.g. in the `Simple.Animation` module:

Change this:

```elm
type alias Animation =
    Internal.Animation

```

To this:
```elm
type Animation
    = Animation Internal.Animation

```

**Comment by andrewMacmurray** [comment]
Thanks for this @MattiaVerticchio! Just one comment about exposing the internals but happy to merge this once that's fixed 👍

**Comment by MattiaVerticchio** [comment]
> Thanks for this @MattiaVerticchio! Just one comment about exposing the internals but happy to merge this once that's fixed 👍 

I changed the `Animation` type as you suggested, however, I’m not entirely sure of the result since it’s my first time using opaque types 😅

File: src/Simple/Animation.elm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module Simple.Animation exposing
-    ( Animation, Millis, fromTo, steps, empty
+    ( Animation(..), Millis, fromTo, steps, empty
```
Comment (andrewMacmurray) [inline src/Simple/Animation.elm:2]: Ah I see the confusion now, doing `Animation(..)` here actually makes it not opaque by exposing it to users of the package - you might be better off doing this in the `Internal.Animation` module so you'd have something like:

```elm
module Internal.Animation exposing
    ( Animation(..)
    , .....
    )

--------


type Animation
    = Animation Animation_


type alias Animation_ =
    { duration : Millis
    , frames : List Frame
    , options : List Option
    }

```

And then in `Simple.Animation`

```elm
module Simple.Animation exposing
    ( Animation
    , .....
    )

-----

type alias Animation =
    Internal.Animation

```

I'm happy to make the change if you don't have time for it though

**Comment by andrewMacmurray** [comment]
> > Thanks for this @MattiaVerticchio! Just one comment about exposing the internals but happy to merge this once that's fixed 👍
> 
> I changed the `Animation` type as you suggested, however, I’m not entirely sure of the result since it’s my first time using opaque types 😅

Thanks for having a go at the change, a bit more moving around and it should be good. There's a great section on the [elm-package guide](https://package.elm-lang.org/help/design-guidelines) about why opaque types are good, particularly for libraries (The section is called `Keeping tags and record constructors secret`)

**Comment by andrewMacmurray** [comment]
Closing this as implemented in #28 - there was a subtle bug with combining X,Y and XY transforms so this PR accounts for that

**Pull request closed without merging**